### ISSUE TITLE
Regexp segfault -->  ("X"x3529) =~ /( (?: \\. | [^\$] ){1,4000} )/gx;

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#6326]
Migrated from [rt.perl.org#21333](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=21333) (status was 'resolved')

Searchable as RT21333$

**Comment by p5pRT** [comment]
### From @aigan

#### Created by @aigan



This row causes a segfault&#8203;:

&nbsp;  \("X"x3529\) =~ /\( \(?&#8203;: \\\\\. | \[^\\$\] \)\{1\,4000\} \)/gx;


I came upon this using the Template Toolkit

&nbsp;    Template&#8203;::Parser\,v 2\.66 \(from TT 2\.08\) segfaults

The regexp is found on row 402 in that module\.


Remove the 'x' and the program works\.

Let the string be shorter than 3529 bytes\, and the program works\.


I'm using Perl 5\.8\.0\-15\, Debian package\.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.8.0:

Configured by Debian Project at Mon Feb 17 13:30:42 UTC 2003.

Summary of my perl5 (revision 5.0 version 8 subversion 0) configuration:
  Platform:
    osname=linux, osvers=2.4.19, archname=i386-linux-thread-multi
    uname='linux cyberhq 2.4.19 #1 smp sun aug 4 11:30:45 pdt 2002 i686 unknown unknown gnulinux '
    config_args='-Dusethreads -Duselargefiles -Dccflags=-DDEBIAN -Dcccdlflags=-fPIC -Darchname=i386-linux -Dprefix=/usr -Dprivlib=/usr/share/perl/5.8.0 -Darchlib=/usr/lib/perl/5.8.0 -Dvendorprefix=/usr -Dvendorlib=/usr/share/perl5 -Dvendorarch=/usr/lib/perl5 -Dsiteprefix=/usr/local -Dsitelib=/usr/local/share/perl/5.8.0 -Dsitearch=/usr/local/lib/perl/5.8.0 -Dman1dir=/usr/share/man/man1 -Dman3dir=/usr/share/man/man3 -Dman1ext=1 -Dman3ext=3perl -Dpager=/usr/bin/sensible-pager -Uafs -Ud_csh -Uusesfio -Uusenm -Duseshrplib -Dlibperl=libperl.so.5.8.0 -Dd_dosuid -des'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=define use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=define d_sfio=undef uselargefiles=define usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='cc', ccflags ='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64',
    optimize='-O3',
    cppflags='-D_REENTRANT -D_GNU_SOURCE -DDEBIAN -fno-strict-aliasing'
    ccversion='', gccversion='3.2.2', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=1234
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=12
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=8
    alignbytes=4, prototype=define
  Linker and Libraries:
    ld='cc', ldflags =' -L/usr/local/lib'
    libpth=/usr/local/lib /lib /usr/lib
    libs=-lgdbm -ldb -ldl -lm -lpthread -lc -lcrypt
    perllibs=-ldl -lm -lpthread -lc -lcrypt
    libc=/lib/libc-2.3.1.so, so=so, useshrplib=true, libperl=libperl.so.5.8.0
    gnulibc_version='2.3.1'
  Dynamic Linking:
    dlsrc=dl_dlopen.xs, dlext=so, d_dlsymun=undef, ccdlflags='-rdynamic'
    cccdlflags='-fPIC', lddlflags='-shared -L/usr/local/lib'

Locally applied patches:
    


@INC for perl v5.8.0:
    /etc/perl
    /usr/local/lib/perl/5.8.0
    /usr/local/share/perl/5.8.0
    /usr/lib/perl5
    /usr/share/perl5
    /usr/lib/perl/5.8.0
    /usr/share/perl/5.8.0
    /usr/local/lib/site_perl
    .


Environment for perl v5.8.0:
    HOME=/home/jonas
    LANG=sv_SE
    LANGUAGE (unset)
    LC_ALL=sv_SE
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=/home/jonas/bin:/usr/local/bin:/usr/bin:/bin:/usr/bin/X11:/usr/games
    PERL_BADLANG (unset)
    SHELL=/bin/bash

```
</details>

**Comment by p5pRT** [comment]
### From @rspier

This is likely because your operating stack size limt is set too low\. 
For bash/sh/ksh/zsh use 'ulimit \-a' to display\.  In tcsh/csh use 'limit'\.

**Comment by p5pRT** [comment]
@rspier - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @aigan

Robert \(via RT\) \<perlbug\-followup@&#8203;perl\.org> writes&#8203;:

> This is likely because your operating stack size limt is set too
> low\.  For bash/sh/ksh/zsh use 'ulimit \-a' to display\.  In tcsh/csh
> use 'limit'\.

jonas@&#8203;jonas&#8203;:\~$ perl \-e'\("X"x3999\) =~ /\( \(?&#8203;: \\\\\. | \[^\\$\] \)\{1\,4000\} \)/gx'
Segmenteringsfel
jonas@&#8203;jonas&#8203;:\~$ ulimit \-a
core file size        \(blocks\, \-c\) 0
data seg size         \(kbytes\, \-d\) unlimited
file size             \(blocks\, \-f\) unlimited
max locked memory     \(kbytes\, \-l\) unlimited
max memory size       \(kbytes\, \-m\) unlimited
open files                    \(\-n\) 1024
pipe size          \(512 bytes\, \-p\) 8
stack size            \(kbytes\, \-s\) unlimited
cpu time             \(seconds\, \-t\) unlimited
max user processes            \(\-u\) 2559
virtual memory        \(kbytes\, \-v\) unlimited
jonas@&#8203;jonas&#8203;:\~$



The first numer vary from place to place\. On this computer it was
higher than 3529\.

"Segmenteringsfel" is swedish för "Segmentation fault"\.

\-\- 
/ Jonas  \-  http&#8203;://jonas\.liljegren\.org/myself/en/index\.html

**Comment by p5pRT** [comment]
### From @smpeters

> \[jonas@&#8203;paranormal\.se \- Fri Feb 21 13&#8203;:34&#8203;:14 2003\]&#8203;:
> 
> 
> This is a bug report for perl from jonas@&#8203;paranormal\.se\,
> generated with the help of perlbug 1\.34 running under perl v5\.8\.0\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> This row causes a segfault&#8203;:
> 
>   \("X"x3529\) =~ /\( \(?&#8203;: \\\\\. | \[^\\$\] \)\{1\,4000\} \)/gx;
> 
> 
> I came upon this using the Template Toolkit
> 
>     Template&#8203;::Parser\,v 2\.66 \(from TT 2\.08\) segfaults
> 
> The regexp is found on row 402 in that module\.
> 
> 
> Remove the 'x' and the program works\.
> 
> Let the string be shorter than 3529 bytes\, and the program works\.
> 
> 
> I'm using Perl 5\.8\.0\-15\, Debian package\.
> 

This appears to have been fixed with change \#27598\.

steve@&#8203;kirk&#8203;:\~/smoke/perl\-current$ perl \-wle'\("X"x3529\) =~ /\( \(?&#8203;: \\\\\. |
\[^\\$\] \)\{1\,4000\} \)/gx;'
Segmentation fault
steve@&#8203;kirk&#8203;:\~/smoke/perl\-current$ \./perl \-wle'\("X"x3529\) =~ /\( \(?&#8203;: \\\\\. |
\[^\\$\] \)\{1\,4000\} \)/gx;'
steve@&#8203;kirk&#8203;:\~/smoke/perl\-current$

**Comment by p5pRT** [comment]
@smpeters - Status changed from 'open' to 'resolved'
